FoodHaus_App/project-swift/Controller/User/EditViewController.swift

```diff
@@ -38,9 +38,17 @@ class EditViewController: UIViewController {
                                                                  "name": name,
                                                                  "phone": phone,
                                                                  "address": address])
-     
-            dismiss(animated: true, completion: nil)
             
+            
+            _ = navigationController?.popViewController(animated: true)
+
+            
+            // cant present info immediately !!!!!!
+//            dismiss(animated: true, completion: nil)
+             // will present info immediately !!!!!!
+//            let vc = self.storyboard?.instantiateViewController(withIdentifier: "Profile")
+//            self.present(vc!, animated: true, completion:  nil)
```

Comment:
Hey Bo, don't use this line `self.present(vc!, animated: true, completion:  nil)`
Use this one instead: `self.navigationController!.pushViewController(vc!, animated: true)`
The navigation bar should appear.

Reply:
You're right! This func is going to present next vc directly, but not dismiss the current one, which means the current vc will be stack. However, I use _ = navigationController?.popViewController(animated: true), when I click save button, in profile page, the info is not updated immediately. Still trying to fix it

Reply:
_ = navigationController?.popViewController(animated: true)
^ this one doesn't seem like a right approach to me, because it doesn't display the changes immediately